Sources/Plasma/FeatureLib/pfConsole/pfConsoleCommands.cpp

```diff
@@ -2431,27 +2431,20 @@ PF_CONSOLE_CMD(App,
 PF_CONSOLE_CMD(App,
                SetLanguage,
                "string language",
-               "Set the language (English, French, German, Spanish, Italian, or Japanese)")
+               "Set the language")
 {
     if (pfConsole::GetPipeline())
     {
         PrintString("This command must be used in an .ini file");
         return;
     }
 
-    if (stricmp(params[0], "english") == 0)
-        plLocalization::SetLanguage(plLocalization::kEnglish);
-    else if (stricmp(params[0], "french") == 0)
-        plLocalization::SetLanguage(plLocalization::kFrench);
-    else if (stricmp(params[0], "german") == 0)
-        plLocalization::SetLanguage(plLocalization::kGerman);
-    else if (stricmp(params[0], "spanish") == 0)
-        plLocalization::SetLanguage(plLocalization::kSpanish);
-    else if (stricmp(params[0], "italian") == 0)
-        plLocalization::SetLanguage(plLocalization::kItalian);
-    else if (stricmp(params[0], "japanese") == 0)
-        plLocalization::SetLanguage(plLocalization::kJapanese);
-
+    for (const auto &lang : plLocalization::GetAllLanguages()) {
+        if (plLocalization::GetLanguageName(lang).compare_i(params[0]) == 0) {
```

Comment:
How would we feel about ISO language codes (i.e., `en`, `fr`, `de`, `it`, `es`, etc.) rather that names? That would open us to support region-specific localizations (such as spelling "Neighbourhood" properly in `en-GB`)

Although I guess that complicates the .loc files too, eh?

Reply:
Ideally we would use standard language codes everywhere instead of custom names, yeah. Migrating the existing .loc files would be quite painful though - it would be a big incompatible change to all existing content, which would break open PRs and require changes to other tools like Korman.